O/ORTools/build_tarballs.jl

```diff
@@ -1,18 +1,25 @@
 # Note that this script can accept some limited command-line arguments, run
 # `julia build_tarballs.jl --help` to see a usage message.
-using BinaryBuilder
+using BinaryBuilder, Pkg
 
 name = "ORTools"
-version = v"9.11.1"
+version = v"9.12.0"
 
 # Collection of sources required to build this package
 sources = [
     GitSource("https://github.com/google/or-tools.git",
-              "f99e8e981e13035bb7ef074776525ad275182ef6")
+              "b8e881fbde473a9e33e0dac475e498559eb0459d")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
+# OR-Tools 9.12 starts requiring CMake 3.24+.
+wget -q "https://cmake.org/files/v3.31/cmake-3.31.0-linux-x86_64.sh"
+chmod a+x cmake-3.31.0-linux-x86_64.sh
+./cmake-3.31.0-linux-x86_64.sh --prefix=/usr/local/ --skip-license
+rm cmake-3.31.0-linux-x86_64.sh
+export PATH=/usr/local/bin/:$PATH
```

Comment:
Why not using `CMake_jll` as host depdency (which contrary to this installation method can be cached)?

Reply:
Apparently, this installation of CMake doesn't work, I can't do `cmake --version` on FreeBSD (https://buildkite.com/julialang/yggdrasil/builds/18478#01956e72-c43b-46f7-bd4a-441fe26f4e19). The change seems similar to what others do to have a recent version of CMake (like https://github.com/dourouc05/Yggdrasil/blob/5ca97400bc9d37ef30ad4e5b58cc711cb1f01cf7/C/Catalyst/build_tarballs.jl, among the simple scripts that update CMake).